### ISSUE TITLE
The game “梅沢由香里のやさしい囲碁” can't load the save_data

### ISSUE THREAD

**Issue by curatorjin** [hrydgard/ppsspp#12360]
### What happens?
When I played "梅沢由香里のやさしい囲碁" on v1.9
I can save my game only once, that means I can not load the saved game or overwrite it.
The game prompts me :"the memstick has changed,can not load/write data"
BTW: I tried to play it on my android cell phone, it works well.

### What *should* happen?
The saved data loaded well.

### What hardware, operating system, and PPSSPP version?  On desktop, GPU matters for graphical issues.
- System Info：
    - Windows 10(x64)
    - NVIDA GeForce MX150
    - i7-8550U
- PPSSPP version: v1.9

> I think there is sth wrong with the path?

**Comment by curatorjin** [comment]
If necessary, I can upload the game iso

**Comment by hrydgard** [comment]
That's weird - could be something with filename characters that are not valid on Windows I guess, if there are some that work on FAT32 filesystems but are still not allowed in Windows. Or we're just mishandling unicode in filenames or something.

**Comment by hrydgard** [comment]
If you look at the ppsspp/memstick/savedata folder and compare with the one you get on Android, do you see any differences in the filenames?

**Comment by curatorjin** [comment]
> If you look at the ppsspp/memstick/savedata folder and compare with the one you get on Android, do you see any differences in the filenames?

They're the same.
dir name "ULJM05384SYS001"

**Comment by hrydgard** [comment]
what are the names of the files inside the dir then? you could take a screenshot or copy the filenames here. Just want to see if there's even a chance that it's unicode problems

**Comment by curatorjin** [comment]
> what are the names of the files inside the dir then? you could take a screenshot or copy the filenames here. Just want to see if there's even a chance that it's unicode problems

![未标题-1](https://user-images.githubusercontent.com/42287407/65773271-8b3a0b80-e16e-11e9-99c4-c88a03746373.jpg)
DATA.BIN
ICON0.PNG
PARAM.SFO

**Comment by hrydgard** [comment]
Okay, thanks, then it's not that. It's a common thing to look for when something works on Android but not on Windows...

**Comment by curatorjin** [comment]
[cvn-uyi.zip](https://github.com/hrydgard/ppsspp/files/3663073/cvn-uyi.zip)

I uploaded the game iso file in case you need it.
(maybe only this game has the problem)

**Comment by hrydgard** [comment]
Please don't upload copyrighted files, for all sorts of legal reasons, sorry.

**Comment by curatorjin** [comment]
oh....I'm really SORRY for uploading that...

**Comment by unknownbrackets** [comment]
It might help if we compare a save that should be identical between both platforms.

What if you start a new game, save as early as possible, and then do that again on the other device.  Then zip both save folders and attach them here.

The biggest other difference between Windows and Android is file truncation.  If the files are a different size, it probably means that's the cause.

-[Unknown]

**Comment by curatorjin** [comment]
> It might help if we compare a save that should be identical between both platforms.
> 
> What if you start a new game, save as early as possible, and then do that again on the other device. Then zip both save folders and attach them here.
> 
> The biggest other difference between Windows and Android is file truncation. If the files are a different size, it probably means that's the cause.
> 
> -[Unknown]

OK，I think I can upload the savedata from different platforms
[Android.zip](https://github.com/hrydgard/ppsspp/files/3667627/Android.zip)
[Windows.zip](https://github.com/hrydgard/ppsspp/files/3667628/Windows.zip)

**Comment by curatorjin** [comment]
I found this game on my old PC, which I used long time ago.
So I tried to load the old data, it WORKS!!!!
The version of ppsspp is 0.9.9.1
So.... I think this issue can be closed?

**Comment by LunaMoo** [comment]
@curatorjin could you explain what you did on windows to get that error? Did you by chance loaded a savestate before trying to load a savedata?
Because by the message that game shows as noted above:
"the memstick has changed,can not load/write data"
To me it would just be another duplicate of #7244 aka user time travel by savestate abuse and not a real bug, Loading savedata without savestates ofc solves that problem.

**Comment by curatorjin** [comment]
> @curatorjin could you explain what you did on windows to get that error? Did you by chance loaded a savestate before trying to load a savedata?
> Because by the message that game shows as noted above:
> "the memstick has changed,can not load/write data"
> To me it would just be another duplicate of #7244 aka user time travel by savestate abuse and not a real bug, Loading savedata without savestates ofc solves that problem.

I just open the game → create a save → load the save → error
I did not save a savestate or do anything about it.
As other games don't have this problem... I think I can just play this game on version 0.9.9.1 to avoid this error.

**Comment by nassau-tk** [comment]
It was tested by my environment.
PPSSPP Ver. : 1.9.3-147
OS: Android 9.0
DeviceName: SONY SO-02K / Xperia XZ1 Compact (Japan)
CPU: Snapdragon835 (Octa-core)
GPU: Adreno 540
Resolution: 720×1280
RAM: 4GB
ROM:32GB
ExternalStrage: microSDXC 512GB

1.Start the game. (The state that I have no save data)
2.Create a "New system data"(=Save data)
3.Saving
4.Loading
  Indeed reading and writing is done.
5.Shut down the game.
6.Restart the game.
7.Load save data done.

By reference.

**Comment by unknownbrackets** [comment]
So that means it's working for you okay, right?

Perhaps this somehow happens on Windows only?

-[Unknown]

**Comment by nassau-tk** [comment]
>>unknownbrackets

Yes,that's right.
I don't have a issue on save & load in android.

**Comment by unknownbrackets** [comment]
Sorry, I think I got this mixed up with another issue before.

The Android and Windows files are slightly different in length, but they also contain different content, so it's hard to tell if this is a red herring.  Maybe it's related to truncation or concurrent file access, but that doesn't seem very likely.

It might help to confirm if this happens or doesn't on Linux or macOS.  But I'm guessing only Windows is affected.

If you put the save files created by Android on Windows, is it able to load?

What about the opposite - what if you put the bad Windows files on Android, is it happy with them then?

-[Unknown]

**Comment by nassau-tk** [comment]
@unknownbrackets 
Sorry,,,,,l'm not useing Windows PPSSPP.
I don't have a time for use PC in my house.
Why? Because,My daughter and son don't permit it!!!
Haha...

”Smart phone is great device I think.”

**Comment by nassau-tk** [comment]
@unknownbrackets 
@hrydgard 
@LunaMoo 
@curatorjin 

Hello,I tested on the Windows 10 (64-bit).
Because, I'm working in home by Covid-19.
Japan is also serious.

That's setting aside....

Mission : Android PPSSPP Save data move to Windows PPSSPP.
Result : The Windows can load & save the Android savedata.

And,I tested clean start.(Delete tha savedata and start.)
Result : Not happen the issue. I can create new savedata and load the savedata. 

PPSSPP Ver.：1.9.3-656
OS：Windows10 (64-bit) Japanese
CPU：Ryzen7 3800X（8-core）
GPU：RADEON RX580
Resolution：2560×1440
RAM：16GB

**Comment by unknownbrackets** [comment]
Hm.  Is that a clean start even in a brand new session of PPSSPP?  Does that mean the issue was potentially fixed by something else?

-[Unknown]

**Comment by nassau-tk** [comment]
@unknownbrackets 
Sorry,I tested only 1.9.3-656 at Windows(64-bit).
"Clean start" meanings "deleted the savedata". 
@curatorjin wrote just "ver.1.9" .
So,I don't know comparative which version of PPSSPP.

It wasn't reproduced by my environment.
I can say only this things....

**Comment by nassau-tk** [comment]
@unknownbrackets 
I tried testing on other Windows10 PC(64-bit).
**Clean Start**

PPSSPP: v1.9.3-656
CPU:i7-4770K
RAM:16GB
OS:Windows10 Home

Result : Not happen the issue.

You should close this topic, I think.

**Comment by unknownbrackets** [comment]
Thanks for testing.  I'm going to close this indeed - sounds like it's working everywhere now.

-[Unknown]
